src/protagonist/API/Features/OriginStrategies/CustomerOriginStrategiesController.cs

```diff
@@ -0,0 +1,204 @@
+    /// Sample request:
+    ///     PUT: /customers/1/originStrategies/68a8931b-e815-492b-bfe9-2f8135ba4898
+    ///     {
+    ///          "regex": "http[s]?://(.*).my-regex.com",
+    ///          "order": "1",
+    ///          "strategy": "basic-http-authentication", 
+    ///          "credentials": "{ \"user\": \"my-username\", \"password\": \"my-password\" }",
+    ///          "optimised": "false"
+    ///     }
+    /// </remarks>
+    [HttpPut]
+    [Route("{strategyId}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> PutCustomerOriginStrategy(
+        [FromRoute] int customerId,
+        [FromRoute] string strategyId,
+        [FromBody] CustomerOriginStrategy strategyChanges,
+        [FromServices] HydraCustomerOriginStrategyValidator validator,
+        CancellationToken cancellationToken)
+    {
+        var validationResult = await validator.ValidateAsync(strategyChanges, 
+            strategy => strategy.IncludeRuleSets("default"), cancellationToken);
+        if (!validationResult.IsValid)
+        {
+            return this.ValidationFailed(validationResult);
+        }
+        
+        OriginStrategyType? strategyType = null;
+        
+        if (strategyChanges.OriginStrategy.HasText())
+        {
+            strategyType = IsStrategyAllowed(strategyChanges.OriginStrategy);
+            if(strategyType == null)
+                return this.HydraProblem($"'{strategyChanges.OriginStrategy}' is not allowed as an origin strategy type", null, 
+                    400, "Invalid origin strategy");
+        } 
+
+        var request = new UpdateCustomerOriginStrategy(customerId, strategyId)
+        {
+            Regex = strategyChanges.Regex,
+            Strategy = strategyType,
+            Credentials = strategyChanges.Credentials,
+            Order = strategyChanges.Order,
+            Optimised = strategyChanges.Optimised
+        };
+        
+        return await HandleUpsert(request, 
+            s => s.ToHydra(GetUrlRoots().BaseUrl),
+            errorTitle: "Failed to update origin strategy",
+            cancellationToken: cancellationToken);
+    }
+    
+    /// <summary>
+    /// Delete a specified origin strategy owned by the user
+    /// </summary>
+    [HttpDelete]
+    [Route("{strategyId}")]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> DeleteCustomerOriginStrategy(
+        [FromRoute] int customerId,
+        [FromRoute] string strategyId)
+    {
+        var deleteRequest = new DeleteCustomerOriginStrategy(customerId, strategyId);
+
+        return await HandleDelete(deleteRequest);
+    }
+    
+    private OriginStrategyType? IsStrategyAllowed(string strategy)
```

Comment:
I think this could be renamed `Is...Allowed` implies a boolean, but it returns an `OriginStrategyType`